aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -215,13 +215,24 @@ module aptos_framework::staking_contract {
     #[view]
     /// Return the number of pending distributions (e.g. commission, withdrawals from stakers).
     ///
-    /// This errors out the staking contract with the provided staker and operator doesn't exist.
+    /// This errors out if the staking contract with the provided staker and operator doesn't exist.
     public fun pending_distribution_counts(staker: address, operator: address): u64 acquires Store {
         assert_staking_contract_exists(staker, operator);
         let staking_contracts = &borrow_global<Store>(staker).staking_contracts;
         pool_u64::shareholders_count(&simple_map::borrow(staking_contracts, &operator).distribution_pool)
     }
 
+    #[view]
+    /// Return the amount that the staker will receive next time distribute() is called.
+    ///
+    /// This errors out if the staking contract with the provided staker and operator doesn't exist.
+    public fun distributable_amount(staker: address, operator: address): u64 acquires Store {
+        assert_staking_contract_exists(staker, operator);
+        let store = borrow_global_mut<Store>(staker);
+        let staking_contract = simple_map::borrow(&store.staking_contracts, &operator);
+        pool_u64::balance(&staking_contract.distribution_pool, staker)
```

Comment:
Haven't read through the changes in `vesting.move` but this doesn't look right. `staking_contract.distribution_pool` might not be up-to-date if there is new inactive. So `distribute` always checks if there is new inactive then [update_distribution_pool](https://github.com/aptos-labs/aptos-core/blob/f3db990e9ae1ad80627f64017b018434a3c9bffc/aptos-move/framework/aptos-framework/sources/staking_contract.move#L595).

Reply:
Thanks for the pointer! I have updated the code to take this into account now